src/services/deepLinkHandler.ts

```diff
@@ -0,0 +1,14 @@
+import {App} from "@capacitor/app"
+import {Nip55Service} from "./Nip55"
+
+export function initializeDeepLinkHandler() {
+  App.addListener("appUrlOpen", (data: {url: string}) => {
+    console.log("Deep link opened", data.url)
+
+    if (data.url.startsWith("nostr+")) {
+      Nip55Service.handleNip55Intent(data.url)
```

Comment:
Will this actually work? My understanding is that browsers aren't able to listen for intents. But maybe there's a shim in the library you're using?

Reply:
You're  right that standard web browsers can't directly listen for intents. However, we're leveraging Capacitor's capabilities here, which bridges the gap between web and native functionality.

In our Capacitor-based app, the `App.addListener('appUrlOpen', ...)` method actually taps into native deep linking mechanisms on both iOS and Android. This allows our app to respond to custom URL schemes like "nostr+" when it's running or launched via a deep link.
also this approach doesn't cover all deep linking scenarios, particularly when the app is completely closed on mobile devices. For those cases, we'll need to implement additional platform-specific code in our native projects.

If you are unsure of this library 
I could go back redo this with
https://github.com/IT-MikeS/capacitor-intents

Reply:
@staab 
also can you assign the issue to me I almost forget to mention it

Reply:
Looks like someone else beat you to the punch: https://github.com/coracle-social/coracle/issues/421#issuecomment-2367124851

I encourage you to contribute to their solution if you're interested in continuing work on this!